Added deprecation documentation of shortest path and allShortestPaths (#1561)
neo4j/neo4j-documentation: cypher/cypher-docs/src/docs/dev/deprecations-additions-and-compatibility.asciidoc

MATCH (n)-[r:REL]->(m) SET n=properties(r)
----

a|
label:syntax[]
label:deprecated[]
[source, cypher, role="noheader"]
----
MATCH (a), (b), allShortestPaths((a)-[r]->(b)) RETURN b

MATCH (a), (b), shortestPath((a)-[r]->(b)) RETURN b
----
a|

> `shortestPath` and `allShortestPaths` with fixed length relationships has been deprecated because of redundancy instead use a Match with a limit of 1 or:

Reviewer: I think "redundancy" is a bit to vague if we already provide reason (which I think we do not usually do).

```suggestion
`shortestPath` and `allShortestPaths` without <<cypher-pattern-varlength, variable-length relationship>> have been deprecated because there is no additional information over using a simple `MATCH` with a `LIMIT` of 1 or:
```

or simply leave it out:

```suggestion
`shortestPath` and `allShortestPaths` without <<cypher-pattern-varlength, variable-length relationship>> have been deprecated. Instead, use a `MATCH` with a `LIMIT` of 1 or:
```

> `shortestPath` and `allShortestPaths` without <<cypher-pattern-varlength, variable-length relationship>> have been deprecated. Instead, use a `MATCH` with a `LIMIT` of 1 or:

Reviewer 2: ```suggestion
`shortestPath` and `allShortestPaths` without <<cypher-pattern-varlength, variable-length relationship>> are deprecated. Instead, use a `MATCH` with a `LIMIT` of 1 or:
```

Reviewer: I agree with the tense.
Would you prefer singular over plural though? Because there are two entities being deprecated, right?

Reviewer 2: My bad, plural of course :)
